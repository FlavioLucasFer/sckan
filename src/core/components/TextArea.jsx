import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { uniqueId } from 'lodash';

import Icon from './Icon';

TextArea.propTypes = {
	title: PropTypes.string,
	className: PropTypes.string,
	icon: PropTypes.string,
	length: PropTypes.string,
};

function TextArea(props) {
	const { 
		title, 
		className, 
		icon, 
		length,
		...rest 
	} = props;

	const id = uniqueId('text-area-');

	useEffect(() => {
		M.CharacterCounter.init(document.querySelectorAll('.materialize-textarea'));
	}, []);

	function renderIcon() {
		if (!icon)
			return null;

		return (
			<Icon className="prefix"
				color="grey-text">
				{icon}
			</Icon>
		);
	}

	return (
		<div className={`input-field ${className || ''}`}>
			{renderIcon()}
			<textarea {...rest} 
				id={id} 
				className="materialize-textarea"
				data-length={length}
				maxLength={length} />
			<label htmlFor={id}>{title}</label>
		</div>
	);
}

export default TextArea;
