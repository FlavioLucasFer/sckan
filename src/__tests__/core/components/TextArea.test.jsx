import React from 'react';
import { shallow } from 'enzyme';

import TextArea from 'core/components/TextArea';

describe('Testing TextArea core component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<TextArea />
		);

		expect(wrapper.debug()).toMatchSnapshot();
		
		wrapper.setProps({
			title: 'textarea test',
			className: 'teal',
			icon: 'business',
			length: '255',
		});
		
		expect(wrapper.debug()).toMatchSnapshot();
	});
});