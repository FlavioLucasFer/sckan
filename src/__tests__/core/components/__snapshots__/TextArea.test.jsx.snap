// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TextArea core component should render correctly 1`] = `
"<div className=\\"input-field \\">
  <textarea id=\\"text-area-1\\" className=\\"materialize-textarea\\" data-length={[undefined]} maxLength={[undefined]} />
  <label htmlFor=\\"text-area-1\\" />
</div>"
`;

exports[`Testing TextArea core component should render correctly 2`] = `
"<div className=\\"input-field teal\\">
  <Icon className=\\"prefix\\" color=\\"grey-text\\">
    business
  </Icon>
  <textarea id=\\"text-area-2\\" className=\\"materialize-textarea\\" data-length=\\"255\\" maxLength=\\"255\\" />
  <label htmlFor=\\"text-area-2\\">
    textarea test
  </label>
</div>"
`;
